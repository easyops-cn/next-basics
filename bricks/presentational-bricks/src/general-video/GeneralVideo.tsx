import React from "react";
import cssStyle from "./style.module.css";
import { Link } from "@next-libs/basic-components";

interface GeneralVedioProps {
  source: string;
  videoTitle?: string;
  preview: boolean;
  height: number;
}

export function getTitle(title: string, source: string): string {
  if (title === null) {
    const tmp = source
      .split("/")
      .pop()
      .split(".");
    tmp.pop();
    title = tmp.join(".");
  }
  return title;
}

export function getVideoType(source: string): string {
  const tmp = source.split(".");
  if (tmp.length >= 2) {
    return `video/${tmp.pop()}`;
  } else {
    return "video/mp4";
  }
}

export function GeneralVideo(props: GeneralVedioProps): React.ReactElement {
  const videoType = getVideoType(props.source);
  const title = getTitle(props.videoTitle, props.source);

  return (
    <div>
      {props.preview ? (
        <>
          <div className={cssStyle.title}>{title}</div>
          <a target="_blank" href={props.source} rel="noopener noreferrer">
            <video height={props.height} className={cssStyle.player}>
              <source src={props.source} type={videoType} />
              Your browser does not support the video tag.
            </video>
          </a>
        </>
      ) : (
        <video controls height={props.height} className={cssStyle.player}>
          <source src={props.source} type={videoType} />
          Your browser does not support the video tag.
        </video>
      )}
    </div>
  );
}
