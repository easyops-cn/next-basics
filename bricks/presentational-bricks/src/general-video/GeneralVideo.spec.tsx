import React from "react";
import { shallow } from "enzyme";
import { GeneralVideo, getTitle, getVideoType } from "./GeneralVideo";

describe("GeneralVideo", () => {
  it("should work with preview=true", () => {
    const wrapper = shallow(
      <GeneralVideo height={150} source="a.mp4" preview={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("should work with preview=false", () => {
    const wrapper = shallow(
      <GeneralVideo height={720} source="a.mp4" preview={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("test getTitle", () => {
    expect(getTitle(null, "a.mp4")).toBe("a");
    expect(getTitle("bbb", "a.mp4")).toBe("bbb");
    expect(getTitle(null, "http://127.0.0.1/aaa/bbb/ccc/a.mp4")).toBe("a");
    expect(getTitle("", "http://127.0.0.1/aaa/bbb/ccc/a.mp4")).toBe("");
  });

  it("test getVideoType", () => {
    expect(getVideoType("a.mp4")).toBe("video/mp4");
    expect(getVideoType("/aa/bb/a.mp4")).toBe("video/mp4");
    expect(getVideoType("http://127.0.0.1/aa/b.b/a.mp4")).toBe("video/mp4");
    expect(getVideoType("a")).toBe("video/mp4");
    expect(getVideoType("a.b")).toBe("video/b");
  });
});
