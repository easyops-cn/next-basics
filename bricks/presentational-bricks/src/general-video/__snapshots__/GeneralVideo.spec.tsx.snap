// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralVideo should work with preview=false 1`] = `
<div>
  <video
    className="player"
    controls={true}
    height={720}
  >
    <source
      src="a.mp4"
      type="video/mp4"
    />
    Your browser does not support the video tag.
  </video>
</div>
`;

exports[`GeneralVideo should work with preview=true 1`] = `
<div>
  <div
    className="title"
  />
  <a
    href="a.mp4"
    rel="noopener noreferrer"
    target="_blank"
  >
    <video
      className="player"
      height={150}
    >
      <source
        src="a.mp4"
        type="video/mp4"
      />
      Your browser does not support the video tag.
    </video>
  </a>
</div>
`;
